src/module.c

```diff
@@ -261,6 +267,10 @@ typedef struct RedisModuleBlockedClient {
 static pthread_mutex_t moduleUnblockedClientsMutex = PTHREAD_MUTEX_INITIALIZER;
 static list *moduleUnblockedClients;
 
+static client **moduleTempClients;
+static size_t moduleTempClientCount = 0;
+static size_t moduleTempClientCap = 0;
```

Comment:
please add some comment here to describe that cache, it's purpose and behavior (shrinking).
maybe define the `32` minimum in some `#define` constant?
please also mention limit and cron mechanism in the top comment of the PR. (to be used as commit comment)